File: practice/11/c++/11/11/11.cpp
#include <iostream>
#include <string>
using namespace std;

int main() {
    setlocale(LC_ALL, "Russian");
    double number, numpow, numberes = 1;
    while (cout << "Введите число, степень которого хотите найти:\n" && !(cin >> number)) {
        cin.clear();
        string str;
        getline(cin, str);
        cout << "Пожалуйста, введите корректные данные.\n";
    }
    while (cout << "Введите степень, в которую хотите возвести число:\n" && !(cin >> numpow)) {
        cin.clear();
        string str;
        getline(cin, str);
        cout << "Пожалуйста, введите корректные данные.\n";
    }
    if (number == 0) {
        cout << "Результат возведения в степень равен:\n1.";
    }
    else if (numpow > 0) {
        while (numpow != 0) {
            numberes *= number;
            numpow -= 1;
        }
        cout << "Результат возведения в степень равен:\n" << numberes;
    }
    else if (numpow < 0) {
        while (numpow != 0) {
            numberes *= number;
            numpow += 1;
        }
        cout << "Результат возведения в степень равен:\n" << 1/numberes;
    }
}

File: practice/11/python/11.py
numberes = 1;
print("Введите число, которое хотите возвести в степень")
number = int(input())
print("Введите степень, в которую хотите возвести число:")
numpow = int(input())
if numpow == 0:
    print("Результат возведения в степень равен: 1")
elif numpow > 0:
    while numpow != 0:
        numberes *= number
        numpow -= 1
    print("Результат возведения в степень равен:", numberes)

elif numpow < 0:
    while numpow != 0:
        numberes *= number
        numpow += 1
    print("Результат возведения в степень равен:", 1/numberes)